webapp/controller/ProductDetails.controller.js

```diff
@@ -52,6 +103,196 @@ sap.ui.define([
 					.filter((el) => data?.includes(el.Id))
 					.map(el => el.Name)
 					.join("")
+		},
+
+		onCreateProductPress: function() {
+			const oView = this.getView();
+			const oModel = oView.getModel();
+			const oFormModel = oView.getModel("FormModel");
+			const aProducts = oModel.getProperty("/Products");
+			const nProductId = aProducts.length + 1;
+			const aSelectedSuppliersId = this.byId("idSuppliersMultiComboBox").getSelectedKeys();
+			const aSuppliers = aSelectedSuppliersId.map(el => ({SupplierId: el}));
+			const sAvailability = oFormModel.getProperty("/Availability");
+			const oDateFormat = DateFormat.getDateTimeInstance({
+				pattern: "yyyy-MM-dd'T'HH:mm:ss"
+			});
+			
+			const oInputBinding = this.byId("idNameInput").getBinding("value");
+			const sValue = oFormModel.getProperty("/Name");
+			try {
+				oInputBinding.getType().validateValue(sValue);
+				if(this.byId("idNameInput").getValueState() === "Error") {
+					return new Error()
+				}
+			} catch (oException) {
+				this.byId("idNameInput").setValueState("Error")
+				return
+			}
+		
+			this.byId("idNameInput").setValueState("None");
+			
+			
+			oFormModel.setProperty("/Availability", JSON.parse(sAvailability.toLowerCase()))
+			oFormModel.setProperty("/Id", String(nProductId));
+			oFormModel.setProperty("/Suppliers", aSuppliers);
+		
+			aProducts.push(oFormModel.getProperty("/"))
```

Comment:
```suggestion
			aProducts.push(oFormModel.getData())
```

.getProperty("/") is quite confusing and hard to read and understand